apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -881,31 +881,32 @@ def _perform_interface_port_operations(self, context, port,
         if not net_info:
             return
 
-        if (self._is_nat_enabled_on_ext_net(ext_net) and
-                self._is_edge_nat(net_info)):
+        nat_enabled = self._is_nat_enabled_on_ext_net(ext_net)
+        is_edge_nat = nat_enabled and self._is_edge_nat(net_info)
+        if not nat_enabled or is_edge_nat:
             if self.per_tenant_context:
-                shadow_l3out = self.name_mapper.l3_out(
+                l3out = self.name_mapper.l3_out(
                     context, ext_net_id,
                     openstack_owner=router['tenant_id'])
-                router_tenant = self._get_tenant(router)
             else:
                 # There is exactly one shadow L3Out for all tenants since there
                 # is exactly one VRF for all tenants
-                shadow_l3out = self.name_mapper.l3_out(context, ext_net_id)
-                vrf_info = self._get_tenant_vrf(router['tenant_id'])
-                router_tenant = vrf_info['aci_tenant']
+                l3out = self.name_mapper.l3_out(context, ext_net_id)
+
+            if is_edge_nat:
+                l3out = self._get_shadow_name_for_nat(l3out, is_edge_nat=True)
```

Comment:
nit: Is it needed to set is_edge_nat in the parameters list, given that it meets the condition in the line above?